### PR TITLE
Update dependency versions

### PR DESCRIPTION
**Pull request by shenanigansd**
Update all dependencies
Explicitly updated Sentry to `1.16.0`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -33,7 +33,7 @@ dependencies = [
 
 [project.optional-dependencies]
 gunicorn = ["gunicorn>=20.1"]  # Lowest which supports wsgi_app in config.
-sentry = ["sentry-sdk[falcon]>=1.5.4"]
+sentry = ["sentry-sdk[falcon]>=1.16.0"]
```
Comment (MarkKoz) [inline pyproject.toml:36]: Mind adding a comment explaining that we want this for Falcon 3.0 support and linking to https://github.com/getsentry/sentry-python/pull/1733 ?

**Review by MarkKoz — CHANGES_REQUESTED** [review]
Comment in pyproject.toml for the falcon constraint can be removed now.

Also this can and should be removed https://github.com/python-discord/snekbox/blob/main/snekbox/utils/logging.py#L27-L28 I looked at the sentry PR and confirmed this warning will no longer be shown since it will try to import the new name first and only import the deprecated one as a fall back.

File: pyproject.toml (hunk 1 above)
Comment (shenanigansd) [reply pyproject.toml]: a7a5ff8f7fc919727acffe084622f98036c2abf6

**Review by MarkKoz — APPROVED** [review]
Thanks for taking care of this!

**Pull request merged**